lib/swtpmtest.pm

```diff
@@ -0,0 +1,115 @@
+# Copyright (C) 2020-2021 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+#
+# Summary: Base module for swtpm test cases
+# Maintainer: rfan1 <richard.fan@suse.com>
+
+package swtpmtest;
+
+use base Exporter;
+use Exporter;
+use strict;
+use warnings;
+use testapi;
+use utils;
+
+our @EXPORT = qw(
+  start_swtpm_vm
+  stop_swtpm_vm
+  swtpm_verify
+);
+
+my $image_path  = '/var/lib/libvirt/images';
+my $guestvm_xml = {
+    swtpm_1 => {
+        xml_file    => {uefi => 'swtpm_uefi_1_2.xml', legacy => 'swtpm_legacy_1_2.xml'},
+        version     => '1.2',
+        ssh_script  => 'ssh_script_1',
+        result_file => '/tmp/swtpm_1',
+    },
+    swtpm_2 => {
+        xml_file    => {uefi => 'swtpm_uefi_2_0.xml', legacy => 'swtpm_legacy_2_0.xml'},
+        version     => '2.0',
+        ssh_script  => 'ssh_script_2',
+        result_file => '/tmp/swtpm_2',
+    },
+};
+
+my $sample_cfg = {
+    uefi   => {vm_name => 'SLES15SP3-UEFI-swtpm',   sample_file => 'swtpm_uefi.xml'},
+    legacy => {vm_name => 'SLES15SP3-LEGACY-swtpm', sample_file => 'swtpm_legacy.xml'},
```

Comment:
These 2 vm are fixed for SLE15SP3, right? will the field need to be updated in the future? (SP4...etc, SLE16...etc.)

Reply:
good question, these img files are located at RMT server, they are SLES15SP3 now, then we can build my own new img files if we need test new guest. 

I am not sure it is a "right" way, but for now, it is simple way for me. do let me know if you have any suggestions

Reply:
let me remove this "hard" version

Reply:
Currently I think it is fine for our testing since we cannot put it on openQA pool :p. BTW, what do you mean the "hard" version ?